functional/FunctionalTest.java

```diff
@@ -2232,6 +2233,73 @@ public static void setBucketPolicy_test4() throws Exception {
   }
 
   /**
+   * Test list bucket policies.
+   */
+  public static void testListBucketPolicies(Map<String, PolicyType> expectedPolicies) throws Exception {
+    long startTime = System.currentTimeMillis();
+    try {
+      for (final Map.Entry<String, PolicyType> policyTypeEntry : expectedPolicies.entrySet()) {
+        client.setBucketPolicy(bucketName, policyTypeEntry.getKey(), policyTypeEntry.getValue());
+      }
+
+      final Map<String, PolicyType> actualPolicies = client.listBucketPolicies(bucketName, "");
+      for (final Map.Entry<String, PolicyType> expectedPolicy : expectedPolicies.entrySet()) {
+        final String expectedPrefix = String.format("%s/%s*", bucketName, expectedPolicy.getKey());
+        final PolicyType expectedPolicyType = expectedPolicy.getValue();
+        if (!actualPolicies.containsKey(expectedPrefix)) {
+          throw new Exception(String.format("policy type not found for objectPrefix %s", expectedPrefix));
+        }
+        final PolicyType actualPolicyType = actualPolicies.get(expectedPrefix);
+        if (!actualPolicyType.equals(expectedPolicyType)) {
+          throw new Exception(
+              String.format(
+                  "found policy type %s for objectPrefix %s; expected %s",
+                  actualPolicyType,
+                  expectedPrefix,
+                  expectedPolicyType
+              )
+          );
+        }
+      }
+
+      for (final String objectPrefix : expectedPolicies.keySet()) {
+        client.removeObject(bucketName, objectPrefix);
+      }
+
+      mintSuccessLog("listBucketPolicies(String bucketName, String objectPrefix)", null, startTime);
+    } catch (Exception e) {
+      mintFailedLog("listBucketPolicies(String bucketName, String objectPrefix)", null, startTime,
+          null, e.toString() + " >>> " + Arrays.toString(e.getStackTrace()));
+      throw e;
+    }
+  }
+
+  /**
+   * Test: single policy type: listBucketPolicies(String bucketName, String objectPrefix).
+   */
+  public static void listBucketPolicy_test1() throws Exception {
+    if (!mintEnv) {
+      System.out.println("Test: listBucketPolicies(String bucketName, String objectPrefix) (Single-Policy Case)");
+    }
+    final Map<String, PolicyType> policyTypeMap = ImmutableMap.of("list-policies-read-only", PolicyType.READ_ONLY);
+    testListBucketPolicies(policyTypeMap);
+  }
+
+  /**
+   * Test: multiple policy types: listBucketPolicies(String bucketName, String objectPrefix).
+   */
+  public static void listBucketPolicy_test2() throws Exception {
+    if (!mintEnv) {
+      System.out.println("Test: listBucketPolicies(String bucketName, String objectPrefix) (Multi-Policy Case)");
+    }
+    final Map<String, PolicyType> policyTypeMap = ImmutableMap.of(
+        "list-policies-write-only", PolicyType.WRITE_ONLY,
+        "list-policies-read-write", PolicyType.READ_WRITE
+    );
+    testListBucketPolicies(policyTypeMap);
+  }
```

Comment:
Below code is much simpler and complete
```java
  /**
   * Test get bucket all policy.
   */
  public static void testGetBucketPolicies(Map<String, PolicyType> policyMap) throws Exception {
    long startTime = System.currentTimeMillis();

    String bucketName = getRandomName();
    client.makeBucket(bucketName);
    try {
      policyMap.forEach((objectPrefix, policy) -> client.setBucketPolicy(bucketName, objectPrefix, policy));
      Map<String, PolicyType> actualPolicies = client.getBucketPolicy(bucketName, "");

      Map<String, PolicyType> expectedPolicies = new HashTable<String, PolicyType>();
      policyMap.forEach((objectPrefix, policy) -> expectedPolicies.put(objectPrefix+"*", policy));

      if (!actualPolicies.equals(expectedPolicies)) {
        throw new Exception("[FAILED] policy list differs");
      }

      mintSuccessLog("getBucketPolicy(String bucketName)", null, startTime);
    } catch (Exception e) {
      ErrorResponse errorResponse = null;
      if (e instanceof ErrorResponseException) {
        ErrorResponseException exp = (ErrorResponseException) e;
        errorResponse = exp.errorResponse();
      }

      // Ignore NotImplemented error
      if (errorResponse != null && errorResponse.errorCode() == ErrorCode.NOT_IMPLEMENTED) {
        mintIgnoredLog("getBucketPolicy(String bucketName)", null, startTime);
      } else {
        mintFailedLog("getBucketPolicy(String bucketName)", null, startTime, null,
                      e.toString() + " >>> " + Arrays.toString(e.getStackTrace()));
      }
      throw e;
    } finally {
      client.removeBucket(bucketName);
    }
  }

  /**
   * Test: single policy type: getBucketPolicy(String bucketName).
   */
  public static void getBucketPolicy_test5() throws Exception {
    if (!mintEnv) {
      System.out.println("Test: getBucketPolicy(String bucketName) (Single-Policy Case)");
    }

    Map<String, PolicyType> policyMap = new HashTable<String, PolicyType>();
    policyMap.put("list-policies-read-only", PolicyType.READ_ONLY);
    testGetBucketPolicies(policyMap);
  }

  /**
   * Test: multiple policy types: getBucketPolicy(String bucketName).
   */
  public static void getBucketPolicy_test6() throws Exception {
    if (!mintEnv) {
      System.out.println("Test: getBucketPolicy(String bucketName) (Multi-Policy Case)");
    }

    Map<String, PolicyType> policyMap = new HashTable<String, PolicyType>();
    policyMap.put("list-policies-write-only", PolicyType.WRITE_ONLY);
    policyMap.put("list-policies-read-write", PolicyType.READ_WRITE);
    testGetBucketPolicies(policyMap);
  }
```

Reply:
Would love to use lambdas, but the project currently has source compatibility set to 1.7. Mind if I bump it to 1.8 in the `build.gradle` file?

Reply:
Yes. You can change `build.gradle`